src/functionaltest/java/com/ericsson/ei/frontend/SubscriptionHandlingFunctionality.java

```diff
@@ -45,20 +50,68 @@ public void testSubscription() throws Exception {
         assert (new WebDriverWait(driver, 10)
                 .until((webdriver) -> subscriptionPage.presenceOfHeader(subscriptionHeaderID)));
 
-        // // Press "Reload" button and verify that two subscriptions with names
-        // "Subscription1" and "Subscription2" are present
+        // Press "Reload" button without enabling LDAP and verify that two subscriptions
+        // with names
+        // "Subscription1" and "Subscription2" are present AND there exists "edit" and
+        // "delete buttons" for unauthorized user "ABCD"
         String response = this.getJSONStringFromFile(RELOAD_TEST_FILE_PATH);
+        String deletePath = "//tr[td[contains(.,'Subscription1')]]/td/button[contains(text(),'Delete')]";
+        String editPath = "//tr[td[contains(.,'Subscription1')]]/td/button[contains(text(),'Edit')]";
+        String viewPath = "//tr[td[contains(.,'Subscription1')]]/td/button[contains(text(),'View')]";
         subscriptionPage.clickReload(response);
         assert (new WebDriverWait(driver, 10)
                 .until((webdriver) -> ((driver.getPageSource().contains("Subscription1")))));
         assert (new WebDriverWait(driver, 10)
                 .until((webdriver) -> ((driver.getPageSource().contains("Subscription2")))));
 
+        assert (subscriptionPage.buttonExist(deletePath) == true);
+        assert (subscriptionPage.buttonExist(editPath) == true);
+        assert (subscriptionPage.buttonExist(viewPath) == true);
+
+        // Given LDAP is enabled, "Reload" subscriptions and then click subscription
+        // page with LDAP enabled with unauthorized user names
+        // Verify that subscriptions exists but only with "View" button
+        String responseSub = this.getJSONStringFromFile(RELOAD_TEST_FILE_PATH_LDAP);
+        String responseAuth = "{\"security\":true}";
+        subscriptionPage.clickReloadLDAP(responseSub, responseAuth);
+        indexPageObject.clickSubscriptionPage();
+        assert (new WebDriverWait(driver, 10)
+                .until((webdriver) -> ((driver.getPageSource().contains("Subscription1")))));
+        assert (subscriptionPage.buttonExist(deletePath) == false);
+        assert (subscriptionPage.buttonExist(editPath) == false);
+        assert (subscriptionPage.buttonExist(viewPath) == true);
+
+        // Given LDAP is enabled, "Reload" subscriptions and then click subscription
+        // page with LDAP enabled with both unauthorized and unauthorized user names (in
+        // this case authorized user is "ABCD" with subscriptions, "subscription1" and
+        // "subscription2")
+        // Verify that current user can see only their own subscriptions' edit and
+        // delete buttons.
+        String key = "currentUser";
```

Comment:
key doesnt say much when it is used later on. I think you should rename this to something like "currentUser" or "currentUserKey" to give a more clear explanation of what that variable actually contains.